src/main/java/Parser.java

```diff
@@ -0,0 +1,169 @@
+package duke;
+
+import duke.Alias;
+import duke.error.DukeException;
+import duke.command.Command;
+import duke.command.AddCommand;
+import duke.command.UpdateCommand;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.ExitCommand;
+import duke.command.ListAlias;
+import duke.command.AddAlias;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Deals with making sense of the user command.
+ * */
+public class Parser {
+
+    /**
+     * Parses given user's input and instantiate to Command object.
+     *
+     * @param fullcommand String input from user
+     * @return Command object
+     */
+    public static Command parse(String fullcommand) throws DukeException {
+        try {
+            String subcommand = fullcommand.split(" ")[0];
+            String command = fullcommand.replaceFirst(subcommand, "").trim();
+            String type;
+            if (Alias.aliases.containsKey(subcommand)) {
+                type = Alias.aliases.get(subcommand);
+            } else {
+                throw new DukeException("       OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
```

Comment:
Hello! Our group reviewed your code during our tutorial. Your parse method seems too long - perhaps you could abstract the splitting of the command and the matching of the aliases to another method called by parse()? Cheers!